### PR TITLE
Refactor end to end tests

### PR DESCRIPTION
**Pull request by stuartmcalpine**
Split the end to end tests up a bit, make cleaner.

### CODE REVIEW

**Comment by JoanneBogart** [comment]
Some of the code changes (to create_registry_db.py and execution.py)  nothing to do with testing as far as I can tell.  They seem to have to do with proper handling of dependencies (same-schema versus production).  
And it appears, roughly speaking, that the old file test_end_to_end_python_api.py has been split into many pieces.
Is that the gist of those changes?  And can you summarize what the intent is of the changes to the files under `unit_tests`?

**Comment by stuartmcalpine** [comment]
> Some of the code changes (to create_registry_db.py and to execution.py) have nothing to do with testing as far as I can tell. They seem to have to do with proper handling of dependencies (same-schema versus production).
> And it appears, roughly speaking, that the old file test_end_to_end_python_api.py has been split into many pieces.
> Is that the gist of those changes? And can you summarize what the intent is of the changes to the files under `unit_tests`?

Yes you are right, I forgot I made some bug fixes when I was refactoring the tests I spotted when looking at the output more closely. I've removed those two bug fixes into their own branch and removed the changes from this branch.

Essentially the changes are slitting the end-to-end tests into many files and now have much of the connection utility in a base file they can all call upon, which will make them more readable and easier to add new tests.

The unit test changes is making use of pytest `parameterize` to make some tests cleaner (and easier to add more variations for the tests). Nothing thats being tested here has changed.

File: tests/unit_tests/test_connection.py [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
+    "config_file,set_env_var",
+    [
+        (_DEFAULT_LOCATION, False),
+        (None, False),
+        (None, True),
+    ],
+)
+def test_connection(config_file, set_env_var):
+    """
+    Test various ways of passing the connection config file
```
Comment (JoanneBogart) [inline tests/unit_tests/test_connection.py:20]: Since all three cases boil down to the same thing - using the file at ${HOME}/.config_reg_access - there is no way to tell if the code in DbConnection is really following a different path other than looking at the output from the verbose setting.  How about creating a file (could have the same contents) in a different location for each case and eliminating other two, so for each call either it finds the file in the specified location or fails?

**Review by JoanneBogart — APPROVED** [review]
I found one typo and there is one, somewhat more substantial suggestion which would take a little work to implement.   It would be good to do it but it's not urgent.
